core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -64,26 +61,11 @@ long allocatedWithPossibleDelta() {
 
     // releasing the memory back to the OS, freeing the block, an opposite of allocation, not thread safe
     void clean() {
-        Field cleanerField = null;
-        try {
-            cleanerField = buffer.getClass().getDeclaredField("cleaner");
-        } catch (NoSuchFieldException e) {
-            e.printStackTrace();
-        }
-        assert cleanerField != null;
-        cleanerField.setAccessible(true);
-        Cleaner cleaner = null;
-        try {
-            cleaner = (Cleaner) cleanerField.get(buffer);
-        } catch (IllegalAccessException e) {
-            e.printStackTrace();
-        }
-        assert cleaner != null;
-        cleaner.clean();
+        UnsafeUtils.unsafe.freeMemory(memAddress);
```

Comment:
Liran had a suggestion that this transfer to another way of memory release may have influenced the memory consumption. This is a bit speculation, as all what `freeMemory()` does' is just releasing memory back to OS. But we can check it by commenting this out and looking on the **on-heap** memory consumption.

Reply:
did not work :(